src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Item/EnableAzureRmRecoveryServicesBackupProtection.cs

```diff
@@ -260,7 +261,55 @@ public override void ExecuteCmdlet()
                         IPsBackupProvider psBackupProvider = (Item != null) ?
                             providerManager.GetProviderInstance(Item.WorkloadType, Item.BackupManagementType)
                             : providerManager.GetProviderInstance(Policy.WorkloadType);
+                                        
+                        if (Policy.WorkloadType == Models.WorkloadType.AzureFiles)
+                        {
+                            // if item & policy are not null and both have ids
+                            AzureFileShareItem afsItem = (Item != null) ? (AzureFileShareItem)Item : null;
+                     
+                         
+                            if (afsItem != null && Policy != null && afsItem.PolicyId != null && Policy.Id != null)
+                            {
+                                Dictionary<UriEnums, string> keyValueDict = HelperUtils.ParseUri(afsItem.PolicyId);
+                                string oldPolicyName = HelperUtils.GetPolicyNameFromPolicyId(keyValueDict, afsItem.PolicyId);
+
+                                keyValueDict = HelperUtils.ParseUri(Policy.Id);
+                                string newPolicyName = HelperUtils.GetPolicyNameFromPolicyId(keyValueDict, Policy.Id);
+
+                                ProtectionPolicyResource oldPolicy = ServiceClientAdapter.GetProtectionPolicy(
+                                    oldPolicyName,
+                                    vaultName: vaultName,
+                                    resourceGroupName: resourceGroupName);
+                                ProtectionPolicyResource newPolicy = ServiceClientAdapter.GetProtectionPolicy(
```

Comment:
please move AFS specific validations to AFS provider

Reply:
@hiaga Is it possible to use ShouldContinue()/ShouldProcess() outside of cmdlet execution block?